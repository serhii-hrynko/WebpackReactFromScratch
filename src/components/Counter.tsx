import React, { useState } from 'react';

import styles from '../assets/styles/counter.module.css';

export const Counter: React.FC = () => {
    const [count, setCount] = useState<number>(0);

    return (
        <div className={styles.counterWrapper}>
            Counter
            <div className={styles.counterContent}>
                <button onClick={() => setCount((state) => state + 1)}>Increment</button>
                <button onClick={() => setCount((state) => state - 1)}>Decrement</button>
                <code>{count}</code>
            </div>
        </div>
    );
};

export default Counter;
