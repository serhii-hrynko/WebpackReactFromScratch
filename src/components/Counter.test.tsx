import React from 'react';
import { act, create } from 'react-test-renderer';
import Counter from './Counter';

describe('components/Counter.tsx', () => {
    test('Counter snapshot', () => {
        const component = create(<Counter />);

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Counter unit', () => {
        const component = create(<Counter />);

        const codeElement = component.root.findByType('code');
        const getCodeElementText = () => codeElement.children[0];

        expect(getCodeElementText()).toBe('0');

        const incrementButton = component.root.find((node) => node.children[0] === 'Increment');
        act(() => {
            incrementButton.props.onClick();
        });
        expect(getCodeElementText()).toBe('1');

        const decrementButton = component.root.find((node) => node.children[0] === 'Decrement');
        act(() => {
            decrementButton.props.onClick();
            decrementButton.props.onClick();
        });
        expect(getCodeElementText()).toBe('-1');
    });
});
