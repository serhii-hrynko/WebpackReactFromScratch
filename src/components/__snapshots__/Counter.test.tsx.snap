// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Counter.tsx Counter snapshot 1`] = `
<div
  className="counterWrapper"
>
  Counter
  <div
    className="counterContent"
  >
    <button
      onClick={[Function]}
    >
      Increment
    </button>
    <button
      onClick={[Function]}
    >
      Decrement
    </button>
    <code>
      0
    </code>
  </div>
</div>
`;
